ibm/data_source_ibm_database.go

```diff
@@ -0,0 +1,546 @@
+	}
+
+	var instance models.ServiceInstance
+
+	if len(filteredInstances) > 1 {
+		return fmt.Errorf(
+			"More than one resource instance found with name matching [%s]\nIf not specified please specify more filters like resource_group_id if instance doesn't exists in default group, location or database", name)
+	}
+	instance = filteredInstances[0]
+
+	d.SetId(instance.ID)
+	d.Set("tags", instance.Tags)
+	d.Set("name", instance.Name)
+	d.Set("status", instance.State)
+	d.Set("resource_group_id", instance.ResourceGroupID)
+	d.Set("parameters", flatmap.Flatten(instance.Parameters))
+	d.Set("location", instance.RegionID)
+
+	serviceOff, err := rsCatRepo.GetServiceName(instance.ServiceID)
+	if err != nil {
+		return fmt.Errorf("Error retrieving service offering: %s", err)
+	}
+
+	d.Set("service", serviceOff)
+
+	servicePlan, err := rsCatRepo.GetServicePlanName(instance.ServicePlanID)
+	if err != nil {
+		return fmt.Errorf("Error retrieving plan: %s", err)
+	}
+	d.Set("plan", servicePlan)
+
+	icdClient, err := meta.(ClientSession).ICDAPI()
+	if err != nil {
+		return fmt.Errorf("Error getting database client settings: %s", err)
+	}
+
+	icdId := EscapeUrlParm(instance.ID)
+	cdb, err := icdClient.Cdbs().GetCdb(icdId)
+	if err != nil {
+		if apiErr, ok := err.(bmxerror.RequestFailure); ok && apiErr.StatusCode() == 404 {
+			return fmt.Errorf("The database instance was not found in the region set for the Provider. Specify the correct region in the provider definition. %v", err)
+		}
+		return fmt.Errorf("Error getting database config for: %s with error %s\n", icdId, err)
+	}
+	d.Set("adminuser", cdb.AdminUser)
+	d.Set("version", cdb.Version)
+
+	groupList, err := icdClient.Groups().GetGroups(icdId)
+	if err != nil {
+		return fmt.Errorf("Error getting database groups: %s", err)
+	}
+	d.Set("groups", flattenIcdGroups(groupList))
+	d.Set("members_memory_allocation_mb", groupList.Groups[0].Memory.AllocationMb)
+	d.Set("members_disk_allocation_mb", groupList.Groups[0].Disk.AllocationMb)
+
+	whitelist, err := icdClient.Whitelists().GetWhitelist(icdId)
+	if err != nil {
+		return fmt.Errorf("Error getting database whitelist: %s", err)
+	}
+	d.Set("whitelist", flattenWhitelist(whitelist))
+
+	var connectionStrings []CsEntry
+	//ICD does not implement a GetUsers API. Users populated from tf configuration.
+	tfusers := d.Get("users").(*schema.Set)
+	users := expandUsers(tfusers)
+	user := icdv4.User{
+		UserName: cdb.AdminUser,
+	}
+	users = append(users, user)
+	for _, user := range users {
+		userName := user.UserName
+		csEntry, err := getConnectionString(d, userName, meta)
+		if err != nil {
+			return fmt.Errorf("Error getting user connection string for user (%s): %s", userName, err)
+		}
+		connectionStrings = append(connectionStrings, csEntry)
+	}
+	d.Set("connectionstrings", flattenConnectionStrings(connectionStrings))
+
+	return nil
+}
+
+func dataSourceIBMDatabaseInstanceRead1(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Why do we need tw omethods dataSourceIBMDatabaseInstanceRead and dataSourceIBMDatabaseInstanceRead1?

Reply:
Omitted to remove. Deleted.